org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/CollectionLiterals.java

```diff
@@ -231,9 +311,34 @@
 	 */
 	@Pure
 	public static <K, V> LinkedHashMap<K, V> newLinkedHashMap(Pair<? extends K, ? extends V>... initial) {
-		LinkedHashMap<K, V> result = new LinkedHashMap<K, V>(initial.length);
-		putAll(result, initial);
-		return result;
+		if (initial.length > 0) {
+			LinkedHashMap<K, V> result = new LinkedHashMap<K, V>(capacity(initial.length));
+			putAll(result, initial);
+			return result;
+		}
+		return newLinkedHashMap();
+	}
+	
+	private static int capacity(int initialSize) {
+		if (initialSize < 3)
+			return initialSize + 1;
+		else if (initialSize < Ints.MAX_POWER_OF_TWO)
+			return initialSize + initialSize / 3;
+		else
+			return Integer.MAX_VALUE;
+	}
+
+	/**
+	 * Creates an empty mutable {@link TreeMap} instance.
+	 * 
+	 * @param comparator
+	 *            the comparator that should be used. May be <code>null</code> which indicates that the natural ordering
+	 *            of the keys should be used.
+	 * @return a new {@link TreeMap}
+	 */
+	@Pure
+	public static <K, V> TreeMap<K, V> newTreeMap(Comparator<? super K> comparator) {
```

Comment:
Should we add an overload newTreeMap() where we define K as K extends Comparable<? super K>

Reply:
Sure, but I'd like to merge this PR before my vacation ;-)
Your proposal is about new API, while this PR is about the performance of existing API.

Reply:
👍